version: '3.8'

volumes:
  auth-data:
    driver: local
  problem-data:
    driver: local
  delivery-data:
    driver: local

networks:
  auth-network:
    driver: bridge
  problem-network:
    driver: bridge
  delivery-network:
    driver: bridge

services:
  # Database
  auth-pgsql-database:
    image: postgres:12.4-alpine
    container_name: fastfeet_auth_postgres
    networks:
      - auth-network
    ports:
      - "49165:5432"
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: fastfeet_auth
      LC_ALL: C
    restart: always
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 7s
      timeout: 5s
    volumes:
      - ./data/auth/pgdata:/var/lib/postgresql/data
  
  # Mongo
  auth-mongo-database:
    image: mongo:4.4.5
    container_name: fastfeet_auth_mongo
    networks:
      - auth-network
    ports:
      - "28015:27017"
    environment:
      MONGO_INITDB_ROOT_USERNAME: mongo
      MONGO_INITDB_ROOT_PASSWORD: mongo
      MONGO_INITDB_DATABASE: mongo_auth
    restart: always
    volumes:
      - ./data/auth/mongodata:/var/lib/mongo/data

  # Cache
  auth-redis-cache:
    image: redis:alpine
    container_name: fastfeet_auth_redis
    networks:
      - auth-network
    ports:
      - "6380:6379"
    command: redis-server --appendonly yes
    volumes:
      - ./data/auth/redis:/data

### PROBLEM
  # Database
  problem-pgsql-database:
    image: postgres:12.4-alpine
    container_name: fastfeet_problem_postgres
    networks:
      - problem-network
    ports:
      - "49166:5432"
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: fastfeet_problem
      LC_ALL: C
    restart: always
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 7s
      timeout: 5s
    volumes:
      - ./data/problem/pgdata:/var/lib/postgresql/data
  
  # Mongo
  problem-mongo-database:
    image: mongo:4.4.5
    container_name: fastfeet_problem_mongo
    networks:
      - problem-network
    ports:
      - "28016:27017"
    environment:
      MONGO_INITDB_ROOT_USERNAME: mongo
      MONGO_INITDB_ROOT_PASSWORD: mongo
      MONGO_INITDB_DATABASE: mongo_problem
    restart: always
    volumes:
      - ./data/problem/mongodata:/var/lib/mongo/data

  # Cache
  problem-redis-cache:
    image: redis:alpine
    container_name: fastfeet_problem_redis
    networks:
      - problem-network
    ports:
      - "6381:6379"
    command: redis-server --appendonly yes
    volumes:
      - ./data/problem/redis:/data

### DELIVERY
  # Database
  delivery-pgsql-database:
    image: postgres:12.4-alpine
    container_name: fastfeet_delivery_postgres
    networks:
      - delivery-network
    ports:
      - "49167:5432"
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: fastfeet_delivery
      LC_ALL: C
    restart: always
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 7s
      timeout: 5s
    volumes:
      - ./data/delivery/pgdata:/var/lib/postgresql/data

  # Mongo
  delivery-mongo-database:
    image: mongo:4.4.5
    container_name: fastfeet_delivery_mongo
    networks:
      - delivery-network
    ports:
      - "28017:27017"
    environment:
      MONGO_INITDB_ROOT_USERNAME: mongo
      MONGO_INITDB_ROOT_PASSWORD: mongo
      MONGO_INITDB_DATABASE: mongo_delivery
    restart: always
    volumes:
      - ./data/delivery/mongodata:/var/lib/mongo/data


  # Cache
  delivery-redis-cache:
    image: redis:alpine
    container_name: fastfeet_delivery_redis
    networks:
      - delivery-network
    ports:
      - "6382:6379"
    command: redis-server --appendonly yes
    volumes:
      - ./data/delivery/redis:/data